src/main/java/edu/project2/tripplanner/service/TripServiceImpl.java

```diff
@@ -0,0 +1,120 @@
+import edu.project2.tripplanner.repository.PlaceRepository;
+import edu.project2.tripplanner.repository.TripRepository;
+import edu.project2.tripplanner.repository.UserRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.Random;
+
+@Service
+@RequiredArgsConstructor
+public class TripServiceImpl implements TripService, Message {
+
+    private final TripRepository tripRepository;
+    private final PlaceRepository placeRepository;
+    private final UserRepository userRepository;
+
+    private Long userId;
+    private Long placeId;
+    private Long tripId;
+
+    @Override
+    public void deleteTripById(Long id) {
+        tripRepository.deleteTripById(id);
+    }
+
+    @Override
+    public Trip deletePlaceFromTrip(TripIdDTO tripIdDTO) {
+         userId=tripIdDTO.getUserId();
+         placeId=tripIdDTO.getPlaceId();
+         tripId=tripIdDTO.getTripId();
+        Place place = placeRepository.findById(placeId)
+                .orElseThrow(()->new NotFoundException(String.format(mes4, placeId)));
+        return tripRepository.findTripByIdAndUserId(userId, tripId)
+                .map(t -> {
+                    t.removePlace(place);
+                    placeRepository.save(place);
+                    return tripRepository.save(t);
+                })
+                .orElseThrow(()->new NotFoundException(String.format(mes5, tripId,userId)));
+    }
+
+    @Override
+    public ResponseEntity<Trip> addTrip(Long userId, Trip trip) {
+        userRepository.findById(userId).map(u -> {
+            Trip trip1=createTrip(trip);
+            u.getTripList().add(trip1);
+            trip1.setUser(u);
+            return tripRepository.save(trip1);
+        });
+
+        return ResponseEntity
+                .status(HttpStatus.OK)
+                .build();
+    }
+
+    @Override
+    public Trip editTrip(TripDTO tripDTO) {
+         userId=tripDTO.getUserId();
+         tripId=tripDTO.getTripId();
+        if (!userRepository.existsById(userId) || !tripRepository.existsById(tripId)) {
+            throw new NotFoundException(String
+                    .format(mes5,tripId,userId));
+        }
+        return tripRepository.findTripByIdAndUserId(userId, tripId)
+                .map(t -> tripRepository.save(createTrip(t)))
+                .orElseThrow(()->new NotFoundException(String.format(mes5,tripId,userId)));
+    }
+
+    @Override
+    public List<Trip> findAllByUserId(Long userId) {
+
+        return tripRepository.findAllByUserId(userId);
+    }
+
+    @Override
+    public Trip addPlaceToTrip(TripIdDTO tripIdDTO) {
+         userId=tripIdDTO.getUserId();
+         placeId=tripIdDTO.getPlaceId();
+         tripId=tripIdDTO.getTripId();
+        Place place = placeRepository.findById(placeId)
```

Comment:
do you have a PlaceService? If not - you should have.
And in this service you autowire not place repo, but place service. And logic should be moved to place service. 

This service will go to Place service, and if there are no places found, from Place service throw exception